src/chartserver/controller.go

```diff
@@ -31,13 +39,26 @@ func NewController(backendServer *url.URL) (*Controller, error) {
 	//Use customized reverse proxy
 	proxy := NewProxyEngine(backendServer)
 
+	//Create http client with customized timeouts
+	client := &http.Client{
```

Comment:
How about creating a client for the backend server?  It'll be easier for testing.